File: modules/opla-core/src/main/java/br/otimizes/oplatool/core/learning/mlmodels/MLPLearningModel.java
package br.otimizes.oplatool.core.learning.mlmodels;

public class MLPLearningModel {
    private boolean decay;
    private String hiddenLayers;
    private double learningRate;
    private double momentum;
    private int trainingTime;

    public MLPLearningModel(boolean decay, String hiddenLayers, double learningRate, double momentum, int trainingTime) {
        this.decay = decay;
        this.hiddenLayers = hiddenLayers;
        this.learningRate = learningRate;
        this.momentum = momentum;
        this.trainingTime = trainingTime;
    }

    public boolean isDecay() {
        return decay;
    }

    public void setDecay(boolean decay) {
        this.decay = decay;
    }

    public String getHiddenLayers() {
        return hiddenLayers;
    }

    public void setHiddenLayers(String hiddenLayers) {
        this.hiddenLayers = hiddenLayers;
    }

    public double getLearningRate() {
        return learningRate;
    }

    public void setLearningRate(double learningRate) {
        this.learningRate = learningRate;
    }

    public double getMomentum() {
        return momentum;
    }

    public void setMomentum(double momentum) {
        this.momentum = momentum;
    }

    public int getTrainingTime() {
        return trainingTime;
    }

    public void setTrainingTime(int trainingTime) {
        this.trainingTime = trainingTime;
    }
}

File: modules/opla-front/src/app/mlconfig/mlmodels/MLPLearningModel.ts
import {MachineLearningAlgorithm} from "./MachineLearningAlgorithm";
import {MachineLearningModel} from "./MachineLearningModel";

export class MLPLearningModel extends MachineLearningModel {

  decay: boolean;
  hiddenLayers: string;
  learningRate: string;
  momentum: string;
  trainingTime: string;

  constructor(decay: string, hiddenLayers: string, learningRate: string, momentum: string, trainingTime: string) {
    super(MachineLearningAlgorithm.MLP);
    this.decay = (decay === "true");
    this.hiddenLayers = hiddenLayers;
    this.learningRate = learningRate;
    this.momentum = momentum;
    this.trainingTime = trainingTime;
  }
}
